code/game/machinery/doors/firedoor.dm

```diff
@@ -254,6 +256,14 @@
 			nextstate = null
 			close()
 
+/obj/machinery/door/firedoor/proc/check_holofan()
+	var/turf/loc = get_turf(src)
+	for(var/obj/structure/holosign/barrier/atmos/holo in loc.contents)
+		if(!holo)
+			continue
```

Comment:
what? why would a holosign be on the turf, but also not exist?
also please don't call turfs loc in var names, just gets confusing
can't you do `return !!(locate(/obj/structure/holosign/barrier/atmos/) in turf_var)`?